core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -1173,17 +1227,20 @@ void clear() {
                 geoHashBitsSizeByColIdx.clear();
             }
 
+            LowerCaseCharSequenceHashSet getAddedCols() {
+                return addedCols;
+            }
+
             int getColumnIndex(DirectByteCharSequence colName) {
-                final int colIndex = columnIndexByName.get(colName);
+                final int colIndex = columnIndexByNameUtf8.get(colName);
                 if (colIndex != CharSequenceIntHashMap.NO_ENTRY_VALUE) {
-                    // If this line is not covered by tests, look at MangledUtf8Sink implementation and usage
                     return colIndex;
```

Comment:
This line (found column index) is never hit in test `testDuplicateFieldNonASCII`. This means that column name hash table is based on utf16 names while incoming name is mangled name. That used to be solved by MangledUtf8Sink

Reply:
that is correct.
this is how we populate the map:
columnIndexByNameUtf8.put(tempSink.encodeUtf8(metadata.getColumnName(n)).**toString()**, n);

Reply:
MangledUtf8Sink is back